File: push/src/main/java/com/nihility/service/ForegroundAbility.java
package com.nihility.service;

public class ForegroundAbility implements XMPushServiceListener {
    public final ForegroundHelper foregroundHelper;

    public ForegroundAbility(ForegroundHelper foregroundHelper) {
        this.foregroundHelper = foregroundHelper;
    }

    @Override
    public void created() {
        foregroundHelper.startForeground();
    }

    @Override
    public void destroy() {
        foregroundHelper.stopForegroundNotification();
    }
}

File: push/src/main/java/com/nihility/service/ForegroundAbility.kt
package com.nihility.service

class ForegroundAbility(val foregroundHelper: ForegroundHelper) : XMPushServiceListener {
    override fun created() {
        foregroundHelper.startForeground()
    }

    override fun destroy() {
        foregroundHelper.stopForegroundNotification()
    }
}
